src/devices/src/virtio/block/device.rs

```diff
@@ -123,6 +123,14 @@ impl DiskProperties {
     }
 }
 
+impl Drop for DiskProperties {
+    fn drop(&mut self) {
+        if let Err(e) = self.file.sync_all() {
+            error!("File sync_all error: {:?}", e);
```

Comment:
Please add disk `id` or `file_path` (or both) so users can identify the emulated disk and backing file for which this error occurs.

Reply:
what about also incrementing a metric on this error?